pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -717,3 +724,64 @@ func (c *VSphereController) createStorageClassController() storageclasscontrolle
 	)
 	return storageClassController
 }
+
+func getUserAndPassword(namespace string, secretName string, infra *ocpv1.Infrastructure, configMapLister corelister.ConfigMapLister, secretInformer corev1informers.SecretInformer,
+) (string, string, error) {
+	secret, err := secretInformer.Lister().Secrets(namespace).Get(secretName)
+	if err != nil {
+		return "", "", err
+	}
+
+	vCenterName, err := getvCenterName(infra, configMapLister)
+	if err != nil {
+		return "", "", err
+	}
+
+	// CCO generates a secret that contains dynamic keys, for example:
+	// oc get secret/vmware-vsphere-cloud-credentials -o json | jq .data
+	// {
+	//   "vcenter.xyz.vmwarevmc.com.password": "***",
+	//   "vcenter.xyz.vmwarevmc.com.username": "***"
+	// }
+	// So we need to figure those keys out
+	var usernameKey, passwordKey string
+
+	if len(secret.Data) > 2 {
+		klog.Warningf("CSI driver can only connect to one vcenter, more than 1 set of credentials found for CSI driver")
+	}
+
+	usernameKey = vCenterName + ".username"
+	passwordKey = vCenterName + ".password"
+
+	if usernameKey == "" || passwordKey == "" {
+		return "", "", fmt.Errorf("could not find vSphere credentials in secret %s/%s", secret.Namespace, secret.Name)
+	}
+
+	// Get user and pass from CCO secret
+	return string(secret.Data[usernameKey]), string(secret.Data[passwordKey]), nil
+}
```

Comment:
I converted the hook into this function and moved it to this file.

Reply:
This comment originally referred to `getUserAndPassword`, not `escapeQuotesAndBackslashes`.